### PR TITLE
SaveDuringPlay only saves SaveDuringPlay components

### PR DESCRIPTION
**Pull request by gaborkb**
### Purpose of this PR

https://jira.unity3d.com/browse/CMCL-283
SaveDuringPlay saves everything that is on a vcam's gameobject, which is not desired. 

### Testing status

- [ ] Added an automated test
- [ ] Passed all automated tests
- [x] Manually tested 

### Documentation status

- [x] Updated [CHANGELOG](https://keepachangelog.com/en/1.0.0/)
- [ ] Updated README (if applicable)
- [ ] Commented all public classes, properties, and methods
- [ ] Updated user documentation

### Samples status

- [ ] Re-packed Extras~/CinemachineExamples.unitypackage

### Technical risk

Low

### Package version

[Justification for updating either the patch, minor, or major version according to the [semantic versioning](https://semver.org/spec/v2.0.0.html) rules]

- [ ] Updated package version

### CODE REVIEW

**Comment by gaborkb** [comment]
I saw that CinemachineExtensions don't have [SaveDuringPlay]. I added it for the parent class (CinemachineExtension), so all extensions now [SaveDuringPlay]. However, I saw CinemachineComponentBase does not have [SaveDuringPlay], instead some (possibly all) the subclasses have [SaveDuringPlay].
Should I do the same for extensions?

File: Runtime/Core/CinemachineExtension.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@ namespace Cinemachine
     /// to the vcam, modifying its generated state
     /// </summary>
     [DocumentationSorting(DocumentationSortingAttribute.Level.API)]
+    [SaveDuringPlay]
```
Comment (glabute) [inline Runtime/Core/CinemachineExtension.cs:12]: Don't do this.  It's up to each extension to decide if it wants SaveDuringPlay

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Bugfix: changing targets on a live vcam was misbehaving
 - Bugfix: Framing transposer now handles empty groups
 - Bugfix: Interrupting a transition with InheritPosition enabled was broken
+- Bugfix: SaveDuringPlay only components that have SaveDuringPlay. Previously it saved all components on a gameobject with at least one component with SaveDuringPlay.
```
Comment (glabute) [inline CHANGELOG.md:25]: This is the correct approach - but is it the best implementation?

File: Editor/Utility/SaveDuringPlay.cs [hunk 3]
```diff
@@ -221,24 +221,21 @@ public bool ScanFields(string fullName, MonoBehaviour b)
         /// Recursively scan the MonoBehaviours of a GameObject and its children.
         /// For each leaf field found, call the OnFieldValue delegate.
         /// </summary>
-        public bool ScanFields(GameObject go, string prefix = null)
```
Comment (glabute) [inline Editor/Utility/SaveDuringPlay.cs:224]: I think you should revert this change and go back to your earlier implementation, with this difference: instead of checking whether the MonoBehaviour is a Cinemachine class, check if it has the SaveDuringPlay attribute.

**Review by glabute — CHANGES_REQUESTED** [review]

**Review by glabute — APPROVED** [review]

**Comment by glabute** [comment]
if (attr.GetType().Name.Contains("SaveDuringPlay")) 

seems a little weak as a test.  Probably better to test for the exact type of the attribute.
(I realize you just took this snippet from elsewhere in the file - it should probably be fixed in both places)
